tests/unit_tests/test_server_client.py

```diff
@@ -1438,3 +1439,31 @@ async def test_using_router() -> None:
     )
 
     app.include_router(router)
+
+
+def _is_valid_uuid(uuid_as_str: str) -> bool:
+    try:
+        uuid.UUID(str(uuid_as_str))
+        return True
+    except ValueError:
+        return False
+
+
+@pytest.mark.asyncio
+async def test_invoke_returns_run_id(app: FastAPI) -> None:
+    """Test the server directly via HTTP requests."""
+    async with get_async_test_client(app, raise_app_exceptions=True) as async_client:
+        response = await async_client.post("/invoke", json={"input": 1})
+        run_id = response.json()["metadata"]["run_id"]
+        assert _is_valid_uuid(run_id)
```

Comment:
These tests are kinda lame, since they don't actually match up the proper run ID. They just assert some run ID was returned.

Do you guys know how to mock out the generation of run_ids for a given runnable run? If I could do that, then my tests would be much better. Not sure how to reach into the underlying infra like that tho

Reply:
There's a fake tracer (FakeTracer)- check it it does whats needed, if not we could use mock patch (though that's brittle)

Reply:
Took a quick look in the code, fake tracer won't help here -- i think current test is fine